### PR TITLE
feat!: replace `dockerImagePrefix` with `dockerSidecarImage`

### LINKED ISSUE #21217: Replace dockerImagePrefix with configurable sidecar image
### What would you like Renovate to be able to do?

Allow a configurable/deterministic sidecar image to be configured for self-hosted users.

### If you have any ideas on how this should be implemented, please tell us here.

Remove `dockerImagePrefix` as a config option.

Add `sidecarImage` as a new global config option, defaulting to `containerbase/sidecar:x.y.z`. It should be used instead during binarySource=docker sidecar executions.

No need to migrate - we should declare it as a breaking change.

Also configure a regex manager to keep sidecar updated within the Renovate repo.

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by RahulGautamSingh**
## Changes
- replace `dockerImagePrefix` with `sidecarImage`

## Context
- Closes: https://github.com/renovatebot/renovate/issues/21217

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/util/exec/docker/index.spec.ts [hunk 1]
```diff
@@ -212,7 +212,7 @@ describe('util/exec/docker/index', () => {
     const preCommands = [null as never, 'foo', undefined as never];
     const commands = ['bar'];
     const envVars = ['FOO', 'BAR'];
-    const image = sideCarImage;
+    const image = dockerSidecarImage;
```
Comment (RahulGautamSingh) [inline lib/util/exec/docker/index.spec.ts:215]:
```suggestion
    const image = sideCarImage;
```

File: renovate.json [hunk 2]
```diff
@@ -29,5 +29,15 @@
...
       "enabled": false
     }
+  ],
+  "regexManagers": [
+    {
+      "fileMatch": ["config/options/index.ts"],
+      "matchStrings": [
+        "default: 'ghcr.io/containerbase/sidecar:(?<currentValue>.*)'"
+      ],
+      "depNameTemplate": "containerbase/sidecar",
```
Comment (viceice) [inline renovate.json:39]:
```suggestion
        "default: '(?<depName>ghcr.io/containerbase/sidecar):(?<currentValue>.*)'"
      ]
```
published times are different, so we should same image as renovate source. also made it simpler.

**Review by viceice — COMMENTED** [review]
needs migration, so old prefix gets joined with `sidecar` and assigned to new option to not break existing users.

**Comment by rarkins** [comment]
@viceice let's just make this one breaking? It only applies to self-hosted users

File: lib/modules/manager/bundler/artifacts.spec.ts [hunk 3]
```diff
@@ -37,6 +37,7 @@ const adminConfig: RepoGlobalConfig = {
   localDir: join('/tmp/github/some/repo'),
   cacheDir: join('/tmp/cache'),
   containerbaseDir: join('/tmp/cache/containerbase'),
+  dockerSidecarImage: 'ghcr.io/containerbase/sidecar:8.0.2',
```
Comment (viceice) [inline lib/modules/manager/bundler/artifacts.spec.ts:40]:
```suggestion
  dockerSidecarImage: 'ghcr.io/containerbase/sidecar',
```
i would prefer this to reduce test changes

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**